tools/render-test/vk-swap-chain.cpp

```diff
@@ -0,0 +1,421 @@
+    m_api = deviceQueue->getApi();
+    
+    // Make sure it's not set initially
+    m_format = VK_FORMAT_UNDEFINED;
+
+    Desc desc(descIn);
+
+#if SLANG_WINDOWS_FAMILY
+    const WinPlatformDesc* platformDesc = static_cast<const WinPlatformDesc*>(platformDescIn);
+    _setPlatformDesc(*platformDesc);
+
+    VkWin32SurfaceCreateInfoKHR surfaceCreateInfo = {};
+    surfaceCreateInfo.sType = VK_STRUCTURE_TYPE_WIN32_SURFACE_CREATE_INFO_KHR;
+    surfaceCreateInfo.hinstance = platformDesc->m_hinstance;
+    surfaceCreateInfo.hwnd = platformDesc->m_hwnd;
+
+    SLANG_VK_RETURN_ON_FAIL(m_api->vkCreateWin32SurfaceKHR(m_api->m_instance, &surfaceCreateInfo, nullptr, &m_surface));
+#else
+    const XPlatformDesc* platformDesc = static_cast<const XPlatformDesc*>(platformDescIn);
+    _setPlatformDesc(*platformDesc);
+
+    VkXlibSurfaceCreateInfoKHR surfaceCreateInfo = {};
+    surfaceCreateInfo.sType = VK_STRUCTURE_TYPE_XLIB_SURFACE_CREATE_INFO_KHR;
+    surfaceCreateInfo.dpy = platformDesc->m_display; 
+    surfaceCreateInfo.window = platformDesc->m_window;
+    
+    SLANG_VK_RETURN_ON_FAIL(m_api->vkCreateXlibSurfaceKHR(m_api->m_instance, &surfaceCreateInfo, nullptr, &m_surface));
+#endif
+
+    VkBool32 supported = false;
+    m_api->vkGetPhysicalDeviceSurfaceSupportKHR(m_api->m_physicalDevice, deviceQueue->getQueueIndex(), m_surface, &supported);
+
+    uint32_t numSurfaceFormats = 0;
+    List<VkSurfaceFormatKHR> surfaceFormats;
+    m_api->vkGetPhysicalDeviceSurfaceFormatsKHR(m_api->m_physicalDevice, m_surface, &numSurfaceFormats, nullptr);
+    surfaceFormats.SetSize(int(numSurfaceFormats));
+    m_api->vkGetPhysicalDeviceSurfaceFormatsKHR(m_api->m_physicalDevice, m_surface, &numSurfaceFormats, surfaceFormats.Buffer());
+
+    // Look for a suitable format
+    List<VkFormat> formats;
+    formats.Add(VulkanUtil::getVkFormat(desc.m_format));
+    // HACK! To check for a different format if couldn't be found
+    if (descIn.m_format == Format::RGBA_Unorm_UInt8)
+    {
+        formats.Add(VK_FORMAT_B8G8R8A8_UNORM);
+    }
+
+    for(int i = 0; i < int(formats.Count()); ++i)
+    {
+        VkFormat format = formats[i];
+        if (_indexOf(surfaceFormats, format) >= 0)
+        {
+            m_format =  format;
+        }
+    }
+
+    if (m_format == VK_FORMAT_UNDEFINED)
+    {
+        return SLANG_FAIL;
+    }
+
+    // Save the desc
+    m_desc = desc;
+
+    SLANG_RETURN_ON_FAIL(_createSwapChain());
+
+    m_desc = desc;
+    return SLANG_OK;
+}
+
+void VulkanSwapChain::getWindowSize(int* widthOut, int* heightOut) const
+{
+#if SLANG_WINDOWS_FAMILY
+    auto platformDesc = _getPlatformDesc<WinPlatformDesc>();
+
+    RECT rc;
+    ::GetClientRect(platformDesc->m_hwnd, &rc);
+    *widthOut = rc.right - rc.left;
+    *heightOut = rc.bottom - rc.top;
+#else
+    auto platformDesc = _getPlatformDesc<XPlatformDesc>();
+
+    XWindowAttributes winAttr = {}; 
+    XGetWindowAttributes(platformDesc->m_display, platformDesc->m_window, &winAttr);
```

Comment:
Have you tested any of the Linux Vulkan stuff, or will we be debugging that once we have a proper CI solution for Linux?

Reply:
This might be the breaking point where we have to consider adopting SDL or some other solution for getting a window and event loop up in a cross-platform manner. I've been trying to avoid that for as long as possible to minimize any complexity it would create for our build.